substrate/client/network/sync/src/engine.rs

```diff
@@ -1231,6 +1234,8 @@ where
 								response receiver.",
 						);
 					},
+					//If OutboundFailure is used, 
+					RequestFailure::Network(_) => {debug_assert!(false, "Don't use deprecated Network");},
```

Comment:
This should not be needed. All `CustomOutboundFailure` variants should be covered.

Reply:
Please no blank-matching here, all variants should be covered.

Reply:
Sorry, looked not at the latest commit, all is fine.